enhancements/ingress/admin-configured-ca-bundle.md

```diff
@@ -0,0 +1,314 @@
+the list of all trusted CA certificates (other than the service CA) in `openshift-config` namespace.
+
+```yaml
+apiVersion: v1
+data:
+  ca-bundle.crt: |
+    -----BEGIN CERTIFICATE-----
+    [...]
+    -----END CERTIFICATE-----
+kind: ConfigMap
+metadata:
+  name: admin-ca-bundle
+  namespace: openshift-config
+```
+
+OpenShift Ingress operator during reconciliation would look for ConfigMaps bearing the names`admin-ca-bundle`
+and `service-ca-bundle` in the `openshift-config` and `openshift-ingress` namespaces, respectively, and
+create a ConfigMap with the name `ingress-ca-bundle` in the `openshift-ingress` namespace.
+
+```yaml
+apiVersion: v1
+data:
+  ca-bundle.crt: |
+    -----BEGIN CERTIFICATE-----
+    <service-ca>
+    -----END CERTIFICATE-----
+    -----BEGIN CERTIFICATE-----
+    <admin-ca>
+    -----END CERTIFICATE-----
+kind: ConfigMap
+metadata:
+  name: ingress-ca-bundle
+  namespace: openshift-ingress
+```
+
+OpenShift Ingress operator should create a watcher to observe `admin-ca-bundle` and `service-ca-bundle` for
+any changes and update `ingress-ca-bundle` accordingly.
+
+OpenShift Router currently uses the CA certificate provided as a process argument
+`default-destination-ca-path` or set as an environment variable `DEFAULT_DESTINATION_CA_PATH` as the
+CA certificate for verifying the service's certificate for re-encrypt termination type.
+
+CA certificate bundle is currently made available to the router using the volume mount option at the path
+`/var/run/configmaps/service-ca/service-ca.crt` which will be updated to
+`/var/run/configmaps/ca-trust/ca-bundle.crt` with the `ingress-ca-bundle` content.
+
+OpenShift Router should create a file watcher to observe `/var/run/configmaps/ca-trust/ca-bundle.crt`
+file and reload when the file content has been modified.
+
+### Risks and Mitigations
+
+CA certificates configured could contain invalid PEM data or has duplicate/expired/revoked certificates
+or uses insecure certificate algorithm which could pose risk and cause vulnerability.
+  - CA certificates could be validated and those not meeting the requirements and could potentially disrupt
+    other services' accessibility can be dropped. And the verification of services' certificates signed
+    by the unconsidered certificates would fail and such services will be inaccessible.
+
+### Drawbacks
+
+When a user updates the `ingress-ca-bundle` ConfigMap, its contents are made available to OpenShift Router
+are eventually updated, though a certain [delay](https://kubernetes.io/docs/tasks/configure-pod-container/configure-pod-configmap/#mounted-configmaps-are-updated-automatically)
+is expected for the new changes to be propagated to the pod.
+
+## Design Details
+
+### Open Questions [optional]
+
+N/A.
+
+### Test Plan
+
+- Create routes with re-encrypt termination type with and without destinationCA.
+  - Routes with empty destinationCA should be accessible only
```

Comment:
*nit* This wording and structure feels a bit vague to me, but I assume you are saying `routes with these types of services`, is that what you mean?
```suggestion
  - Routes with empty destinationCA should be accessible only when using:
      - A Service that uses a certificate signed by OpenShift CA
      - A Service that uses a certificate signed by custom CA and is present in `admin-ca-bundle`.
```

Reply:
I had to rebase commit to remove sentences from older commit.